### ISSUE TITLE
System to unaggregate and aggregate a transaction on Schedule E (Line 24) on Form 3X

### ISSUE THREAD

**Issue by GreggMoreland** [fecgov/fecfile-web-app#1381]
Business Reason

As a user, I will be able to "unaggregate" (then with the option to aggregate) an Independent Expenditure (Schedule E, Line 24) transaction so that I can provide additional information for a previous transaction without double-counting the amount for the associated entity.

 Acceptance Criteria

**Given** the user has a Schedule E transaction
**When** the user takes the action to "unaggregate" that transaction
**Then** the transaction amount will _not be included_  in the "per election office sought" aggregate (Calendar YTD) or the entity aggregate (calculated behind the scenes for itemization purposes)
**And** the user can re-aggregate the transaction if needed.

 Wireframes

### QA Notes ### 
 - [x] BA/Design Quality Checklist complete
- #798 already built out Schedule A (This is repetitive for this ticket)
- NOTE to QA testing only Form 3x for this ticket
### DEV Notes ### 
 - In *.fec file, there is nothing to indicate that a transaction is aggregated or not. It's just not added in the aggregation for that entity.
### Design ### 
 null

**Comment by toddlees** [comment]
Waiting on response from FEC about whether clicking unaggregate will have the same impact on itemization as it does on other transactions.  When unaggregating a schedule A, the entity aggregate is impacted and therefore the whether or not it's itemized changes (>$200 rule)
Schedule E has 2 aggregates (1 visible: election, 1 invisible: entity).  Schedule E also has the >$200 itemization rule for the entity aggregate. FEC needs to decide if when the user clicks unaggragate which happens
- the amount is not counted towards the entity aggregate or the election aggregate when it is 'unaggregated'.  Because it's 'unaggregated' it will not count towards itemization rules (current behavior of schedule a)
- the amount is not counted for the election aggregate, but even though it's 'unaggregated', it's counted in the entity aggregate and counts towards itemization rules

**Comment by sasha-dresden** [comment]
Thanks for the update. I currently have it set so you can utilize the same aggregate/unaggregate command that is used by Schedule A. I'll leave this on hold till we have further updates on what to do with the actual action that should take place.

**Comment by AureliaKhorsand** [comment]
@toddlees @sasha-dresden update from Ryan is that the "unaggregate" should apply to BOTH the election and entity aggregates, which will keep it consistent with how it works for Receipts (which just has the entity aggregate). I updated the ACs, let me know if you have further questions.

**Comment by AureliaKhorsand** [comment]
BA note: there is no automatic disaggregation for IE workflows yet. _If_ that is implemented in the future, need to ensure automatic disaggregation works the same way.

**Comment by mjtravers** [comment]
Passes CR. Sending to QA.

**Comment by WiseQA** [comment]
@MitchellTCG @AureliaKhorsand @mjtravers is the unaggregate and aggregate for Schedule E on both Form 3x and Form 24 forms ?   Checking as both forms are currently displaying and wasn't called out which forms this ticket applied to. - Thanks

<img width="1273" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNWlvQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--c1ef04035ed09a5b2c0a8923e5d0a775ea7d6172/image.png" alt="image.png" />

**Comment by AureliaKhorsand** [comment]
@WiseQA confirming that this ticket functionality should only address Form 3X Schedule E. We will address aggregate/unaggregate on Form 24 (whether it's keeping them or taking them out) in a separate ticket once we get more info from FEC. Thank you for catching!

**Comment by WiseQA** [comment]
@AureliaKhorsand thanks for the clarification as to which Form(s).   NOTE to QA testing only Form 3x for this ticket

**Comment by WiseQA** [comment]
During testing met with @AureliaKhorsand for clarification of QA questions.  These are the things needing fixed for this ticket.

1.   Aggregate field should not be displayed on Schedule E.   Remove Aggregate field.

<img width="1176" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeFdwQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--9eea5d4820d6d86bb1952c1101ff3e12a53e54bf/image.png" alt="image.png" />

___________________________________________________________________________________________________________________________
2.  When selecting "unaggregate" from the kabob the transaction type in the table displays "Unitemized" instead of "Unaggregate".   Change verbiage under transaction to "Unaggregated"

<img width="1136" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeG1wQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--c045c2ac9b7db19a17d281dca129aa91c455f510/image.png" alt="image.png" />

____________________________________________________________________________________________________________________
<img width="1085" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeHVwQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--4bcc75cfff1e713065ad66a7f4a0814a01a86b09/image.png" alt="image.png" />

__________________________________________________________________________________________________________________________
3.   When selecting "Aggregate" from the kabob the option for Aggregate should be removed from the kabob and only display Unaggregate as an option.  (Note: the same applies if select Unaggregate then Aggregate should only display)

<img width="1165" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeWlwQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--bdfb7674d121aa6ef17d3a49cb89cdfc2865da01/image.png" alt="image.png" />

________________________________________________________________________________________________________________________
@mjtravers @MitchellTCG returned to in-progress for the above fixes per BA review with @AureliaKhorsand (reference ticket #798)

**Comment by mjtravers** [comment]
The proper code flags were set up to fix the issues found by QA.

Passes CR. Sending back to QA.

**Comment by WiseQA** [comment]
QA review verified fixes during initial QA review were fixed.  (Note:  Tested Form 3x)

1.   Aggregate field removed from Schedule E.

<img width="1069" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOEsyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--2b68b91b6660f85b480afb14bd68e91297334837/image.png" alt="image.png" />

2.  When selecting "unaggregate" from the kabob the transaction type in the table displays "Unaggregated"

<img width="1127" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOFMyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--d8225cd797a9296537a27821f14a48c2ffd9887a/image.png" alt="image.png" />

3.  When selecting "Aggregate" from the kabob the option for Aggregate is removed from the kabob and only display Unaggregate as an option. (Note: the same applies if select Unaggregate then Aggregate should only display)

<img width="1131" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOFcyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--9b9f9f784e57b8db2559dfd46ebd53496a1a98af/image.png" alt="image.png" />

<img width="1132" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOGUyQWc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--cabf49e32e57a3bf6d321d9d826bf23ea41d613d/image.png" alt="image.png" />

QA Review Completed.   Moved to Stage Ready
